test/integration/epp/hermetic_test.go

```diff
@@ -736,6 +736,88 @@ func TestFullDuplexStreamed_KubeInferenceModelRequest(t *testing.T) {
 			),
 			wantMetrics: map[string]string{},
 		},
+		{
+			name: "select active lora with subsetting tag, all pods available",
+			requests: integrationutils.GenerateStreamedRequestSet(
+				logger,
+				"test2",
+				modelSQLLora,
+				[]interface{}{"192.168.1.1:8000", "192.168.1.2:8000", "192.168.1.3:8000"}),
+			// Pod 1 will be picked because it has relatively low queue size, the requested model active, low KV cache, and within subset.
+			pods: newPodStates(
+				podState{index: 0, queueSize: 0, kvCacheUsage: 0.2, activeModels: []string{"foo", "bar"}},
+				podState{index: 1, queueSize: 0, kvCacheUsage: 0.1, activeModels: []string{"foo", modelSQLLoraTarget}},
+				podState{index: 2, queueSize: 10, kvCacheUsage: 0.2, activeModels: []string{"foo", "bar"}},
+			),
+
+			wantMetrics: map[string]string{
+				"inference_model_request_total": inferenceModelRequestTotal([]label{
+					{"model_name", modelSQLLora},
+					{"target_model_name", modelSQLLoraTarget},
+				}),
+			},
+			wantErr: false,
+			wantResponses: integrationutils.NewRequestBufferedResponse(
+				"192.168.1.2:8000",
+				fmt.Sprintf(`{"max_tokens":100,"model":%q,"prompt":"test2","temperature":0}`, modelSQLLoraTarget),
+				&configPb.HeaderValueOption{
+					Header: &configPb.HeaderValue{
+						Key:      "hi",
+						RawValue: []byte("mom"),
+					},
+				},
+			),
+		},
+		{
+			name: "select active lora with subsetting tag, some pods match",
+			requests: integrationutils.GenerateStreamedRequestSet(
+				logger,
+				"test2",
+				modelSQLLora,
+				[]interface{}{"192.168.1.3:8000"}),
+			// Pod 3 will be picked because it is the only one matching subsetting target.
```

Comment:
```suggestion
			// Pod 3 has high queue and kv cache utilization, but it will still be picked because it is the only one matching subsetting target.
```